src/gluonts/torch/model/tide/estimator.py

```diff
@@ -0,0 +1,419 @@
+# permissions and limitations under the License.
+
+from typing import Optional, Iterable, Dict, Any, List
+
+import torch
+import lightning.pytorch as pl
+
+from gluonts.core.component import validated
+from gluonts.dataset.common import Dataset
+from gluonts.dataset.field_names import FieldName
+from gluonts.dataset.loader import as_stacked_batches
+from gluonts.itertools import Cyclic
+from gluonts.model.forecast_generator import DistributionForecastGenerator
+from gluonts.torch.modules.loss import DistributionLoss, NegativeLogLikelihood
+from gluonts.time_feature import (
+    TimeFeature,
+    time_features_from_frequency_str,
+    minute_of_hour,
+    hour_of_day,
+    day_of_month,
+    day_of_week,
+    day_of_year,
+    month_of_year,
+    week_of_year,
+)
+from gluonts.transform import (
+    Transformation,
+    Chain,
+    RemoveFields,
+    SetField,
+    AsNumpyArray,
+    AddObservedValuesIndicator,
+    AddTimeFeatures,
+    AddAgeFeature,
+    VstackFeatures,
+    InstanceSplitter,
+    ValidationSplitSampler,
+    TestSplitSampler,
+    ExpectedNumInstanceSampler,
+    MissingValueImputation,
+    DummyValueImputation,
+    InstanceSampler,
+    InstanceSplitter,
+    SelectFields,
+)
+
+from gluonts.torch.model.estimator import PyTorchLightningEstimator
+from gluonts.torch.model.predictor import PyTorchPredictor
+from gluonts.torch.distributions import (
+    DistributionOutput,
+    StudentTOutput,
+    NormalOutput,
+)
+
+from .lightning_module import TiDELightningModule
+
+PREDICTION_INPUT_NAMES = [
+    "feat_static_real",
+    "feat_static_cat",
+    "past_time_feat",
+    "past_target",
+    "past_observed_values",
+    "future_time_feat",
+]
+
+TRAINING_INPUT_NAMES = PREDICTION_INPUT_NAMES + [
+    "future_target",
+    "future_observed_values",
+]
+
+
+class TiDEEstimator(PyTorchLightningEstimator):
+    """
+    An estimator training the TiDE model form the paper
+    https://arxiv.org/abs/2304.08424 extended for probabilistic forecasting.
+
+    This class is uses the model defined in ``TiDEModel``,
+    and wraps it into a ``TiDELightningModule`` for training
+    purposes: training is performed using PyTorch Lightning's ``pl.Trainer``
+    class.
+
+    Parameters
+    ----------
+    freq
+        Frequency of the data to train on and predict.
+    prediction_length
+        Length of the prediction horizon.
+    context_length
+        Number of time steps prior to prediction time that the model
+        takes as inputs (default: ``4 * prediction_length``).
+    feat_proj_hidden_dim
+        Size of the feature projection layer (default: 40).
+    encoder_hidden_dim
+        Size of the dense encoder layer (default: 40).
+    decoder_hidden_dim
+        Size of the dense decoder layer (default: 40).
+    temporal_hidden_dim
+        Size of the temporal decoder layer (default: 40).
+    distr_hidden_dim
+        Size of the distribution projection layer (default: 40).
```

Comment:
i think you can safely make this dim smaller by default.. since for point forecast it works when this is `1`, so it should also work fine with it say `10` or even `4`

Reply:
@maxc01 what is the number of parameters in the default configuration?

Reply:
the orignal model is not probablilistic

Reply:
i did the same for patchTST too...

Reply:
I mean total number of the model's parameters, in its layers

Reply:
@kashif seems the default hidden dim is too hight... lowered it down to 4.

Reply:
@lostella the number of trainable parameters depends on the context size and the prediction length, there is no such a fixed number, for example, if prediction length=168 and context length is also 168, the trainable parameters is 133K.